datasets/cats_vs_dogs.py

```diff
@@ -1,67 +1,84 @@
-        transforms.Resize((256, 256)),
-        transforms.RandomAffine(degrees=10, translate=(0.05, 0.05), shear=5),
-        transforms.RandomPerspective(distortion_scale=0.3, p=0.2),
-        transforms.CenterCrop((180, 180)),
-        transforms.ColorJitter(brightness=.7),
-        transforms.GaussianBlur(kernel_size=(5, 5), sigma=(0.1, 5)),
-        transforms.RandomHorizontalFlip(),
-        ])
-    return train_transform(orig_img)
-
-
-def augment_blur(orig_img):
+class CatsvsDogs(Dataset):
     """
-    Augment with center crop and bluring
+    `Cats vs Dogs dataset <https://www.kaggle.com/datasets/salader/dogs-vs-cats>` Dataset.
+
+    Args:
+    root_dir (string): Root directory of dataset where ``KWS/processed/dataset.pt``
+        exist.
+    d_type(string): Option for the created dataset. ``train`` or ``test``.
+    transform (callable, optional): A function/transform that takes in an PIL image
+        and returns a transformed version.
+    resize_size(int, int): Width and height of the images to be resized for the dataset.
+    augment_data(bool): Flag to augment the data or not. If d_type is `test`, augmentation is
+        disabled.
     """
-    train_transform = transforms.Compose([
-        transforms.Resize((256, 256)),
-        transforms.CenterCrop((220, 220)),
-        transforms.GaussianBlur(kernel_size=(5, 5), sigma=(0.1, 5))
-        ])
-    return train_transform(orig_img)
 
-
-def catsdogs_get_datasets(data, load_train=True, load_test=True, aug=2):
-    """
-    Load Cats & Dogs dataset
-    """
-    (data_dir, args) = data
-    path = data_dir
-    dataset_path = os.path.join(path, "cats_vs_dogs")
-    is_dir = os.path.isdir(dataset_path)
-    if not is_dir:
+    labels = ['cat', 'dog']
+    label_to_id_map = {k: v for v, k in enumerate(labels)}
+    label_to_folder_map = {'cat': 'cats', 'dog': 'dogs'}
+
+    def __init__(self, root_dir, d_type, transform=None,
+                 resize_size=(128, 128), augment_data=False):
+        self.root_dir = root_dir
+        self.data_dir = os.path.join(root_dir, 'cats_vs_dogs', d_type)
+
+        if not self.__check_catsvsdogs_data_exist():
+            self.__print_download_manual()
+            sys.exit("Dataset not found!")
+
+        self.__get_image_paths()
+
+        self.album_transform = None
+        if d_type == 'train' and augment_data:
+            self.album_transform = album.Compose([
+                album.GaussNoise(var_limit=(1.0, 20.0), p=0.25),
+                album.RGBShift(r_shift_limit=15, g_shift_limit=15, b_shift_limit=15, p=0.5),
+                album.ColorJitter(p=0.5),
+                album.SmallestMaxSize(max_size=int(1.2*min(resize_size))),
+                album.ShiftScaleRotate(shift_limit=0.05, scale_limit=0.05, rotate_limit=15, p=0.5),
+                album.RandomCrop(height=resize_size[0], width=resize_size[1]),
+                album.HorizontalFlip(p=0.5),
+                album.Normalize(mean=(0.0, 0.0, 0.0), std=(1.0, 1.0, 1.0))])
```

Comment:
One question: Doesn't this normalize all images & all channels independently? It is different than dividing the entire dataset by 255, which we employ in the other vision examples. Whichever correct way is needed to be applied to all examples.

Reply:
That's correct. I assumed that it divides each channel with its own maximum value. Doublechecked the documentation and it divides with max possible, which is 255.